src/clip-breaker/MultiClipBreakerSpell.sol

```diff
@@ -0,0 +1,126 @@
+    function list() external view returns (bytes32[] memory);
+    function list(uint256 start, uint256 end) external view returns (bytes32[] memory);
+    function xlip(bytes32 ilk) external view returns (address);
+}
+
+interface ClipperMomLike {
+    function setBreaker(address clip, uint256 level, uint256 delay) external;
+}
+
+interface ClipLike {
+    function wards(address who) external view returns (uint256);
+    function stopped() external view returns (uint256);
+}
+
+contract MultiClipBreakerSpell is DssEmergencySpell {
+    string public constant override description = "Emergency Spell | Multi Clip Breaker";
+    /// @dev During an emergency, set the breaker level to 3  to prevent botyh `kick()`, `redo()` and `take()`.
+    uint256 public constant BREAKER_LEVEL = 3;
+    /// @dev The delay is not applicable for level 3 breakers, so we set it to zero.
+    uint256 public constant BREAKER_DELAY = 0;
+
+    IlkRegistryLike public immutable ilkReg = IlkRegistryLike(_log.getAddress("ILK_REGISTRY"));
+    ClipperMomLike public immutable clipperMom = ClipperMomLike(_log.getAddress("CLIPPER_MOM"));
+
+    event SetBreaker(bytes32 indexed ilk, address indexed clip);
+
+    /**
+     * @notice Sets breakers, when possible, for all Clip instances that can be found in the ilk registry.
+     */
+    function _emeregencyActions() internal override {
+        bytes32[] memory ilks = ilkReg.list();
+        _doSetBreaker(ilks);
+    }
+
+    /**
+     * @notice Sets breakers for all Clips in the batch.
+     * @dev This is an escape hatch to prevent this spell from being blocked in case it would hit the block gas limit.
+     *      In case `end` is greater than the ilk registry length, the iteration will be automatically capped.
+     * @param start The index to start the iteration (inclusive).
+     * @param end The index to stop the iteration (inclusive).
+     */
+    function setBreakerInBatch(uint256 start, uint256 end) external {
+        uint256 maxEnd = ilkReg.count() - 1;
+        bytes32[] memory ilks = ilkReg.list(start, end < maxEnd ? end : maxEnd);
+        _doSetBreaker(ilks);
+    }
+
+    /**
+     * @notice Sets breakers, when possible, for all Clip instances that can be found from the `ilks` list.
+     * @param ilks The list of ilks to consider.
+     */
+    function _doSetBreaker(bytes32[] memory ilks) internal {
+        for (uint256 i = 0; i < ilks.length; i++) {
+            bytes32 ilk = ilks[i];
+            address clip = ilkReg.xlip(ilk);
+
+            if (clip == address(0)) continue;
+
+            try ClipLike(clip).wards(address(clipperMom)) returns (uint256 ward) {
+                // Ignore Clip instances that have not relied on ClipperMom.
+                if (ward == 0) continue;
+            } catch Error(string memory reason) {
+                // If the reason is empty, it means the contract is most likely not a Clip instance.
+                require(bytes(reason).length == 0, reason);
```

Comment:
If for some reason one of the contracts revert with a reason (i.e: wrong auth/ward setup), this spell will revert and fail to break all of them. Since this is to be used in an emergency situation it might be better to break all that are possible and emit events for failures to be checked afterwards.

Reply:
In L78-L86 we try to explicitly avoid all possible failure conditions by skipping them. The example you pointed out is covered by checking if the `ClipperMom` is a ward of the `Clip` contract (L80).
So this call would only fail if the contract does not implement the functions called. If that happens, the revert reason is empty and we can just ignore it.

OTOH, maybe it would be a good idea to log errors with events instead of ignoring them.